crates/api/src/site/mod_log.rs

```diff
@@ -42,69 +53,268 @@ impl Perform for GetModlog {
     ) = if data.community_id.is_none() {
       blocking(context.pool(), move |conn| {
         Ok((
-          ModRemoveCommunityView::list(conn, mod_person_id, page, limit)?,
-          ModBanView::list(conn, mod_person_id, page, limit)?,
-          ModAddView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgePersonView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgeCommunityView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgePostView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgeCommentView::list(conn, mod_person_id, page, limit)?,
+          match type_ {
+            ModlogActionType::All | ModlogActionType::ModBan => ModBanView::list(
+              conn,
+              mod_person_id,
+              other_person_id,
+              page,
+              limit,
+              hide_modlog_names,
+            )?,
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::ModAdd => ModAddView::list(
+              conn,
+              mod_person_id,
+              other_person_id,
+              page,
+              limit,
+              hide_modlog_names,
+            )?,
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::ModRemoveCommunity => {
+              if other_person_id.is_none() {
+                ModRemoveCommunityView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::AdminPurgePerson => {
+              if other_person_id.is_none() {
+                AdminPurgePersonView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::AdminPurgeCommunity => {
+              if other_person_id.is_none() {
+                AdminPurgeCommunityView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::AdminPurgePost => {
+              if other_person_id.is_none() {
+                AdminPurgePostView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::AdminPurgeComment => {
+              if other_person_id.is_none() {
```

Comment:
I believe you can move these if conditions directly into the match statement, like:
```
ModlogActionType::All | ModlogActionType::AdminPurgeComment if other_person_id.is_none() => {
    ...
}
```